src/quicr_client_raw_session.h

```diff
@@ -278,29 +278,32 @@ class ClientRawSession
   qtransport::TransportContextId transport_context_id;
   ClientStatus client_status{ ClientStatus::TERMINATED };
 
-  /*
-   * Nested map to reassemble message fragments
-   *
-   *    Structure:
-   *       fragments[<circular index>] = map[quicr_name] = map[offset] = data
-   *
-   *    Circular index is a small int value that increments from 1 to max. It
-   *    wraps to 1 after reaching max size.  In this sense, it's a circular
-   *    buffer. Upon moving to a new index the new index data will be purged (if
-   *    any exists).
-   *
-   *    Fragment reassembly avoids timers and time interval based checks. It
-   *    instead is based on received data. Every message quicr_name is checked to
-   *    see if it's complete. If so, the published object callback will be
-   *    executed. If not, it'll only update the map with the new offset value.
-   *    Incomplete messages can exist in the cache for as long as the circular
-   *    index hasn't wrapped to the same point in cache.  Under high load/volume,
-   *    this can wrap within a minute or two.  Under very little load, this could
-   *    linger for hours. This is okay considering the only harm is a little extra
-   *    memory being used. Extra memory is a trade-off for being event/message
-   *    driven instead of timer based with threading/locking/...
-   */
-  std::map<int, std::map<quicr::Name, std::map<int, bytes>>> fragments;
+  // Nested map to reassemble message fragments
+  //
+  // Structure:
+  //    fragments[<circular index>] = map[quicr_name] = map[offset] = data
+  //
+  // Circular index is a small int value that increments from 0 to max. It
+  // wraps to 0 after reaching max size.  In this sense, it's a circular
+  // buffer. Upon moving to a new index the new index data will be purged (if
+  // any exists).
+  //
+  // Fragment reassembly avoids timers and time interval based checks. It
+  // instead is based on received data. Every message quicr_name is checked
+  // to see if it's complete. If so, the published object callback will be
+  // executed. If not, it'll only update the map with the new offset value.
+  // Incomplete messages can exist in the cache for as long as the circular
+  // index hasn't wrapped to the same point in cache.  Under high
+  // load/volume, this can wrap within a minute or two.  Under very little
+  // load, this could linger for hours. This is okay considering the only harm
+  // is a little extra memory being used. Extra memory is a trade-off for being
+  // event/message driven instead of timer based with threading/locking/...
+  uint32_t circular_index{ 0 };
+  static constexpr size_t max_pending_per_buffer = 5000;
+  static constexpr size_t max_fragment_buffers = 20;
```

Comment:
This should be configurable. 20 only allows for a little over 24k. Right? We have messages at least up to 80k.  The design was to allow up to 10MB, although we don't go that large.

Reply:
What sort of configurable are you thinking of?  Right now, it's compile-time configurable through these parameters, just like it was before.  Also these constants are the same as before.  I hope you don't mean we need runtime configurability?

The buffer capacity is a little deceptive here.  The `max_pending_per_buffer` is in units of messages; the amount of data in each message is unbounded.  (Maybe it shouldn't be!)  So you can have 10MB messages, as long as you don't have more than 100k in flight at once.  That comes out to 1TBps, so I think we're probably OK :)

Reply:
I think the thought was that we could maybe make this runtime configurable from the client maybe?

Reply:
I don't think we should do that.  To avoid making this massively complex, you would have to make sure that the client only set these values once, and given that we've got plenty of capacity, it doesn't seem worth even the modest complexity of making set-once values.

Reply:
Right, a little hard to see it on my phone. IMO it shouldn't be unbounded. It should be configurable. I was reading the two the other way around, so yeah we have 5000 fragments with only 20 (fixed) in flight. It works for now... but considering all the effort for a refactor it's a bit of a miss tuning wasn't added for runtime. Also, it's a bit of a miss that the refactor didn't include fixing the fixed 1200 byte fragments.